File: week6/pset6/readability/readability.py
from cs50 import get_string


def main():
    # Promt user for text
    text = get_string("Text: ")

    # Calculate letters, words and sentences
    words = 1
    letters = sentences = 0

    for i in text:
        if i.isalnum():
            letters += 1
        elif i.isspace():
            words += 1
        elif i == '.' or i == '!' or i == '?':
            sentences += 1

    # Coleman-Liau formula application
    # L is the average number of letters per 100 words in the text
    # S is the average number of sentences per 100 words in the text
    L = letters / words * 100
    S = sentences / words * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    # Printing out the grade level
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


# Script guard
if __name__ == "__main__":
    main()

File: week2/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    // Prompt user for input
    string text;
    do
    {
        text = get_string("Text: ");
    }
    while (strlen(text) == 0);

    // Counting characters, words, sentences

    int letters = 0, words = 1, sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (isspace(text[i]))
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Calculations based on Coleman-Liau index formula: index = 0.0588 * L - 0.296 * S - 15.8
    // L is the average number of letters per 100 words in the text
    // S is the average number of sentences per 100 words in the text

    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Printing out the output

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}